### DISCUSSION TITLE
Dependant multiple select field loading relationship data instead of options

### DISCUSSION THREAD

**Discussion by nadeaumarc** [filamentphp/filament#9414] [Help] [answered]
In my form, I have a 2 select fields with the second being dependant from the first. 

**My tables:** 

```
Table "fournisseur_stts"
  "id"
  "nom"
  "created_at"
  "updated_at"
  "deleted_at"

Table "action_fournisseur_stt"
  "action_id"
  "fournisseur_stt_id"

Table "actions"
  "id"
  "nom"
  "deleted_at"
```

**Relationships:**

    public function actions(): BelongsToMany
    {
        return $this->belongsToMany(Action::class);
    }

    public function fournisseurStts(): BelongsToMany
    {
        return $this->belongsToMany(FournisseurStt::class);
    }

**Code:**

```
  Select::make('fournisseur_stt_id')
      ->relationship('fournisseurStt', 'nom')
      ->required()
      ->reactive()

  Select::make('Actions')
      ->required()
      ->multiple()
      ->relationship('actions', 'nom')
      ->options(function (callable $get) {
          $fournisseur_stt = FournisseurStt::find($get('fournisseur_stt_id'));

          if (! $fournisseur_stt) {
              return Action::orderBy('nom')->pluck('nom', 'id');
          }
          return $fournisseur_stt->actions->pluck('nom', 'id');
      })
```

The problem is that the 'Actions' select seems to be loading the data from the relationship instead of ->options().

If I remove the ->multiple() or ->relationship(), the ->options() works perfectly. 

How can I make the dependant field work properly?

**Comment by sinnbeck** [comment]
How is it dependent? Just whether it is shown? Maybe you can use this? https://filamentphp.com/docs/3.x/forms/getting-started#dependant-fields

**Reply by nadeaumarc** [reply]
When Select::make('fournisseur_stt_id') is changed, the Select::make('Actions') is supposed to return the Actions related to that fournisseur_stt. For some reason it seems to be returning the data directly from the relationship (all the entries in Actions) instead of options because if I remove ->relationship('actions', 'nom') the condition in the options works but of course won't save properly.

I also tried it with the ->live and Get as you linked and I got the same results as with ->reactive and callable.

It's a many to many relationship so maybe it has something to do with it? (I edited my post and added tables and relationships).

**Reply by VincvanAs** [reply, marked as answer]
Maybe modifyQueryUsing within the relationship could solve your issue.https://filamentphp.com/docs/3.x/forms/fields/select#customizing-the-relationship-option-labels

**Reply by nadeaumarc** [reply]
I tried it out and can't seem to get it to work, I keep getting an error: 

**foreach() argument must be of type array|object, null given**

This is what I used:

```
    ->relationship(
        name: 'actions',
        modifyQueryUsing: function (Builder $query, callable $get) {
                $query->whereHas('fournisseurStts', function ($subquery) use ($get) {
                    $subquery->where('fournisseur_stt_id', $get('fournisseur_stt_id'));
                });
        }
    )
```

**Reply by nadeaumarc** [reply]
Your answer ended up being very helpful, thank you VincvanAs! I ended up modifying a table to make things simpler and realised I was missing _titleAttribute_ before _modifyQueryUsing_. Ended up working perfectly:

```
->relationship(
    name: 'actionsFournisseurStt',
    titleAttribute: 'actions.nom',
    modifyQueryUsing: fn (Builder $query, callable $get) => $query->leftJoin('actions', 'action_fournisseur_stt.action_id', '=', 'actions.id')->select('actions.nom', 'action_fournisseur_stt.id')->orderBy('actions.nom')->where('fournisseur_stt_id', $get('fournisseur_stt_id')),
),
```
